backend/business-partner-agent/src/main/java/org/hyperledger/bpa/impl/prooftemplates/ProofTemplateElementVisitor.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright (c) 2020-2021 - for information on the respective copyright owner
+ * see the NOTICE file and/or the repository at
+ * https://github.com/hyperledger-labs/business-partner-agent
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.hyperledger.bpa.impl.prooftemplates;
+
+import org.hyperledger.aries.api.present_proof.PresentProofRequest;
+import org.hyperledger.bpa.model.*;
+import org.hyperledger.bpa.model.prooftemplate.*;
+
+import java.util.*;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.*;
+import java.util.stream.Stream;
+
```

Comment:
@etschelp  can you review my javadocs, please?

Reply:
Looks good. I think the api of this class is confusing because the Visitor and the Conversion are basically 1:1 related and the visitor needs the conversion class to orchestrate the flow. A future refactoring could be to merge this into one class and then do something like `Visitor.INSTANCE(config).visit(bpaProofTemplate).getResult()`. 

Fixing the access modifiers made that clearer and the only thing that I would add to the class level comment is something like `see {@link ProofTemplateConversion#proofRequestViaVisitorFrom()} for an example on how to use this class`